histolab/tiler.py

```diff
@@ -1121,3 +1121,288 @@ def _scores(
+    ):
+        self.scorer = scorer
+        self.threshold = threshold
+
+        super().__init__(
+            tile_size,
+            level,
+            check_tissue,
+            tissue_percent,
+            pixel_overlap,
+            prefix,
+            suffix,
+            mpp=mpp,
+        )
+
+    def extract(
+        self,
+        slide: Slide,
+        extraction_mask: BinaryMask = BiggestTissueBoxMask(),
+        report_path: str = None,
+        log_level: str = "INFO",
+    ) -> None:
+        """Extract grid tiles and save them to disk, according to a scoring function and
+        following this filename pattern:
+        `{prefix}tile_{tiles_counter}_level{level}_{x_ul_wsi}-{y_ul_wsi}-{x_br_wsi}-{y_br_wsi}{suffix}`
+
+        Save a CSV report file with the saved tiles and the associated score.
+
+        Parameters
+        ----------
+        slide : Slide
+            Slide from which to extract the tiles
+        extraction_mask : BinaryMask, optional
+            BinaryMask object defining how to compute a binary mask from a Slide.
+            Default `BiggestTissueBoxMask`.
+        report_path : str, optional
+            Path to the CSV report. If None, no report will be saved
+        log_level: str, {"DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"}
+            Threshold level for the log messages. Default "INFO"
+
+        Raises
+        ------
+        TileSizeError
+            If the tile size is larger than the slide size
+        LevelError
+            If the level is not available for the slide
+        """
+        level = logging.getLevelName(log_level)
+        logger.setLevel(level)
+        self._validate_level(slide)
+        self.tile_size = self._tile_size(slide)
+        self.pixel_overlap = int(self._scale_factor(slide) * self.pixel_overlap)
+        self._validate_tile_size(slide)
+
+        filtered_tiles_scores, filtered_tiles_scaled_scores = self._tiles_generator(
+            slide, extraction_mask
+        )
+
+        tiles_counter = 0
+        filenames = []
+
+        for tiles_counter, (score, tile_wsi_coords) in enumerate(filtered_tiles_scores):
+            tile = slide.extract_tile(
+                tile_wsi_coords,
+                tile_size=self.final_tile_size,
+                mpp=self.mpp,
+                level=self.level if self.mpp is None else None,
+            )
+            tile_filename = self._tile_filename(tile_wsi_coords, tiles_counter)
+            tile.save(os.path.join(slide.processed_path, tile_filename))
+            filenames.append(tile_filename)
+            logger.info(
+                f"\t Tile {tiles_counter} - score: {score} saved: {tile_filename}"
+            )
+
+        if report_path:
+            self._save_report(
+                report_path,
+                filtered_tiles_scores,
+                filtered_tiles_scaled_scores,
+                filenames,
+            )
+
+        logger.info(f"{tiles_counter+1} Grid Tiles have been saved.")
```

Comment:
why the +1?

Reply:
> why the +1?

Because tiles_counter starts at 0.
If you want to print how many iterations you have done you need to add 1, or am I missing something?

Reply:
what if tiles_counters remains 0 ? It's an edge case but still possible IMHO